Sources/Build/Triple.swift

```diff
@@ -0,0 +1,114 @@
+/*
+ This source file is part of the Swift.org open source project
+
+ Copyright (c) 2014 - 2017 Apple Inc. and the Swift project authors
+ Licensed under Apache License v2.0 with Runtime Library Exception
+
+ See http://swift.org/LICENSE.txt for license information
+ See http://swift.org/CONTRIBUTORS.txt for Swift project authors
+ */
+
+/// Triple - Helper class for working with Destination.target values
+///
+/// Used for parsing values such as x86_64-apple-macosx10.10 into
+/// set of enums. For os/arch/abi based conditions in build plan.
+///
+/// @see Destination.target
+public struct Triple {
+    public let tripleString: String
+
+    public let arch: Arch
+    public let vendor: Vendor
+    public let os: OS
+    public let abi: ABI
+
+    public enum Error: Swift.Error {
+        case badFormat
+        case unknownArch
+        case unknownOS
+    }
+
+    public enum Arch: String {
+        case x86_64
+        case armv7
+    }
+
+    public enum Vendor: String {
+        case unknown
+        case apple
+    }
+
+    public enum OS: String {
+        case darwin
+        case macOS = "macosx"
+        case linux
+
+        fileprivate static let allKnown:[OS] = [
+            .darwin,
+            .macOS,
+            .linux
+        ]
+    }
+
+    public enum ABI: String {
+        case unknown
+        case android = "androideabi"
+    }
+
+    public init(_ string: String) throws {
+        let components = string.split(separator: "-").map { String($0) }
+
+        guard components.count >= 3 && components.count <= 4 else {
```

Comment:
This can just be `components.count == 3`

Reply:
No it is `components.count == 3 || components.count == 4`
May be it is confusing
Will change it